File: authentication/src/main/java/webdoc/authentication/domain/response/ArrayResponse.java
package webdoc.authentication.domain.response;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.List;

/*
 * 배열 응답 객체
 */
@Getter
@RequiredArgsConstructor
public class ArrayResponse<T> {
    private final List<T> data;
    private final int httpStatus;
    private final int size;

    public static <R> ArrayResponse<R> of(List<R> data, int httpStatus){
       return new ArrayResponse<R>(data,httpStatus,data.size());
    }

}

File: socket/response/ArrayResponse.js
// 배열 응답 객체
class ArrayResponse {
    constructor(data, httpStatus) {
      this.data = data;
      this.httpStatus = httpStatus;
      this.size = data.length;
    }
  
    static of(data, httpStatus) {
      return new ArrayResponse(data, httpStatus);
    }
}

module.exports = ArrayResponse;
